packages/web-pkg/src/composables/actions/spaces/useSpaceActionsEditReadmeContent.ts

```diff
@@ -3,20 +3,58 @@ import { computed } from 'vue'
 import { useGettext } from 'vue3-gettext'
 
 import { useOpenWithDefaultApp } from '../useOpenWithDefaultApp'
-import { getRelativeSpecialFolderSpacePath } from '@ownclouders/web-client/src/helpers'
+import { buildSpace, getRelativeSpecialFolderSpacePath } from '@ownclouders/web-client/src/helpers'
 import { useClientService } from '../../clientService'
-import { useUserStore } from '../../piniaStores'
+import { useConfigStore, useSpacesStore, useUserStore } from '../../piniaStores'
+import { useCreateSpace } from '../../spaces'
+import { Drive } from '@ownclouders/web-client/src/generated'
 
 export const useSpaceActionsEditReadmeContent = () => {
   const clientService = useClientService()
   const { openWithDefaultApp } = useOpenWithDefaultApp()
+  const { createDefaultMetaFolder } = useCreateSpace()
   const userStore = useUserStore()
+  const spacesStore = useSpacesStore()
+  const configStore = useConfigStore()
   const { $gettext } = useGettext()
 
   const handler = async ({ resources }: SpaceActionOptions) => {
-    const markdownResource = await clientService.webdav.getFileInfo(resources[0], {
-      path: getRelativeSpecialFolderSpacePath(resources[0], 'readme')
-    })
+    let markdownResource = null
+    try {
+      await clientService.webdav.getFileInfo(resources[0], { path: '.space' })
+    } catch (_) {
+      await createDefaultMetaFolder(resources[0])
+      markdownResource = await clientService.webdav.putFileContents(resources[0], {
+        path: '.space/readme.md'
+      })
+
+      const { data: updatedDriveData } = await clientService.graphAuthenticated.drives.updateDrive(
+        resources[0].id as string,
```

Comment:
I guess this comes from former times.

```suggestion
        resources[0].id,
```